### PR TITLE
image-builder: Add NSDAX_BIN for passing in compiled nsdax tool

### PR DESCRIPTION
**Pull request by crobinso**
In Fedora we are running the osbuilder scripts on the client machine,
to generate an initrd for the running host kernel. In this setup,
there's currently a runtime dependency on gcc for compiling the nsdax
tool, which is suboptimal.

Add NSDAX_BIN environment variable; if specified, image-builder.sh
will use that path as the nsdax tool. This let's ship a compiled
nsdax tool to users and drop the runtime gcc dependency

Fixes: #417

### CODE REVIEW

File: image-builder/image_builder.sh [hunk 1]
```diff
@@ -478,8 +486,11 @@ main() {
 	fi
 
 	if [ -n "$container_engine" ]; then
-		build_with_container "${rootfs}" "${image}" "${fs_type}" "${block_size}" \
-						  "${root_free_space}" "${agent_bin}" "${agent_init}" "${container_engine}"
+		build_with_container "${rootfs}" \
+			"${image}" "${fs_type}" "${block_size}" \
+			"${root_free_space}" "${agent_bin}" \
+			"${agent_init}" "${container_engine}" \
+			"${nsdax_bin}"
```
Comment (devimc) [inline image-builder/image_builder.sh]: ok, I see something wrong here, ${nsdax_bin} is the 9th element, but in `build_with_container` is the 8th (local nsdax_bin="$8"),  could you please add a new variable in `build_with_container` to save the 8th element (local container_engine ="$8") ?

Comment (crobinso) [reply image-builder/image_builder.sh]: Nice catch! Sorry for not testing that path. I've updated the commit now

Comment (devimc) [reply image-builder/image_builder.sh]: thanks @crobinso, I blame @devimc for that bug

**Comment by fidencio** [comment]
@crobinso, cool, really cool.
It'll help **a lot** distros in case they don't want to bring gcc (and friends) as dependency of osbuilder.

**Pull request merged**